pkg/pipeline/source/sdk.go

```diff
@@ -217,37 +217,37 @@ func (s *SDKSource) joinRoom() error {
 }
 
 func (s *SDKSource) awaitRoomTracks() error {
-	var tracks []lksdk.TrackPublication
-
-	for _, p := range s.room.GetRemoteParticipants() {
-		for _, track := range p.TrackPublications() {
-			if s.shouldSubscribe(track) {
-				if err := s.subscribe(track); err != nil {
-					return err
-				}
-
-				tracks = append(tracks, track)
+	// await expected subscriptions
+	expected := 0
+	for _, rp := range s.room.GetRemoteParticipants() {
+		pubs := rp.TrackPublications()
+		for _, pub := range pubs {
+			if s.shouldSubscribe(pub) {
+				expected++
 			}
 		}
 	}
+	if err := s.awaitExpected(expected); err != nil {
+		return err
+	}
 
-	deadline := time.After(subscriptionTimeout)
-	trackCount := len(tracks)
-loop:
-	for i := 0; i < trackCount; i++ {
+	// lock any incoming subscriptions
+	s.subLock.Lock()
+	defer s.subLock.Unlock()
+
+	for {
 		select {
+		// check errors from any tracks published in the meantime
```

Comment:
maybe rename variable to `pub`?

Reply:
guess these are remote pubs and sub here. Got confused by that check. Did not exactly understand what errors this is catching.

Reply:
probably makes sense to rename `s.errors`